xinuk-core/src/main/scala/pl/edu/agh/xinuk/SimulationWithCurvedSmellPropagation.scala

```diff
@@ -0,0 +1,87 @@
+package pl.edu.agh.xinuk
+
+import java.awt.Color
+import java.io.File
+
+import akka.actor.{ActorRef, ActorSystem}
+import akka.cluster.sharding.{ClusterSharding, ClusterShardingSettings}
+import com.typesafe.config.{Config, ConfigFactory, ConfigRenderOptions}
+import com.typesafe.scalalogging.LazyLogging
+import net.ceedubs.ficus.readers.ValueReader
+import pl.edu.agh.xinuk.algorithm.MovesController
+import pl.edu.agh.xinuk.config.{GuiType, XinukConfig, XinukConfigWithBendFactors}
+import pl.edu.agh.xinuk.gui.GuiActor
+import pl.edu.agh.xinuk.model.Grid.CellArray
+import pl.edu.agh.xinuk.model.parallel.{ConflictResolver, Neighbour, NeighbourPosition}
+import pl.edu.agh.xinuk.model._
+import pl.edu.agh.xinuk.simulation.WorkerActor
+
+import scala.collection.immutable.TreeSet
+import scala.util.{Failure, Success, Try}
+
+class SimulationWithCurvedSmellPropagation[ConfigType <: XinukConfig : ValueReader](
```

Comment:
Jeżeli dodatkowe pola konfiguracji dodamy do bazowego `XinukConfig`u, to unikniemy tworzenia całej tej klasy.